ui/src/components/search-box/QasSearchBox.vue

```diff
@@ -101,63 +156,104 @@ export default {
       }
     },
 
-    hasResults () {
-      return !!this.searchResults.length
+    isDisabled () {
+      return (!this.useLazyLoading && !this.list.length) || this.mx_isFetching
+    },
+
+    showEmptyResult () {
+      return this.useEmptySlot && !this.mx_hasFilteredOptions && !this.mx_isFetching
+    },
+
+    showSpinnerDots () {
+      return this.mx_hasFilteredOptions && this.mx_isFetching
+    },
+
+    showInnerLoading () {
+      return !this.mx_hasFilteredOptions && this.mx_isFetching
     }
   },
 
   watch: {
     defaultFuseOptions (value) {
+      if (this.useLazyLoading) return
+
       this.fuse.options = { ...this.fuse.options, ...value }
     },
 
-    hasResults (value) {
+    mx_hasFilteredOptions (value) {
       !value && this.$emit('empty-result')
     },
 
-    list: {
-      handler (value) {
-        this.fuse = new Fuse(value, this.defaultFuseOptions)
+    mx_search: {
+      async handler (value) {
+        this.$emit('update:modelValue', value)
 
-        this.setResults(this.search)
-        this.updateResultsModel(value)
-      },
+        if (this.useLazyLoading) {
+          await this.mx_filterOptionsByStore(value)
 
-      deep: true
+          this.$refs.infiniteScrollRef.resume()
+
+          return
```

Comment:
- poderia utilizar `return this.$refs.infiniteScrollRef.resume()` ?
- remover linha pulada antes do return